pkg/controller/registry/reconciler/grpc.go

```diff
@@ -325,3 +320,28 @@ func (c *GrpcRegistryReconciler) CheckRegistryServer(catalogSource *v1alpha1.Cat
 	healthy = true
 	return
 }
+
+func (c *GrpcRegistryReconciler) swapLabels(updatePod *corev1.Pod, key string) error {
+	updatePod.Labels[CatalogSourceLabelKey] = key
+	updatePod.Labels[CatalogSourceUpdateKey] = ""
+
+	// Update the update pod to promote it to serving pod
+	_, err := c.OpClient.KubernetesInterface().CoreV1().Pods(updatePod.GetNamespace()).Update(context.TODO(), updatePod, metav1.UpdateOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "error creating new pod: %s", updatePod.GetName())
+	}
+
+	return nil
+}
+
+// containerReady returns true if the given Pod has a ready status for its container.
+func containerReady(pod *corev1.Pod) bool {
+	if pod.Status.ContainerStatuses != nil {
+		for _, status := range pod.Status.ContainerStatuses {
+			if status.Ready == true {
+				return true
+			}
+		}
+	}
+	return false
```

Comment:
The existing code obviously works with our current implementation of catalogsources, but it would be nice to be bullet proof in the future.

Reply:
So this pod is the catalog pod, we only spin up one container inside the pod per https://github.com/operator-framework/operator-lifecycle-manager/blob/master/pkg/controller/registry/reconciler/reconciler.go#L88 so the distinction is pretty much a design choice. It's possible in the future to have more than one container in the catalog and this would break, so I can check the suggested way instead.